src/main/java/net/dv8tion/jda/bot/entities/impl/ApplicationInfoImpl.java

```diff
@@ -0,0 +1,152 @@
+/*
+ *     Copyright 2015-2017 Austin Keener & Michael Ritter
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.bot.entities.impl;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import net.dv8tion.jda.bot.entities.ApplicationInfo;
+import net.dv8tion.jda.core.JDA;
+import net.dv8tion.jda.core.Permission;
+import net.dv8tion.jda.core.entities.User;
+
+public class ApplicationInfoImpl implements ApplicationInfo
+{
+    private final JDA api;
+
+    private final String description;
+    private boolean doesBotRequireCodeGrant;
+    private final String iconId;
+    private final String id;
+    private boolean isBotPublic;
+    private final String name;
+    private final User owner;
+
+    public ApplicationInfoImpl(JDA api, String description, boolean doesBotRequireCodeGrant, String iconId, String id,
+            boolean isBotPublic, String name, User owner)
+    {
+        this.api = api;
+        this.description = description;
+        this.doesBotRequireCodeGrant = doesBotRequireCodeGrant;
+        this.iconId = iconId;
+        this.id = id;
+        this.isBotPublic = isBotPublic;
+        this.name = name;
+        this.owner = owner;
+    }
+
+    @Override
+    public final boolean doesBotRequireCodeGrant()
+    {
+        return this.doesBotRequireCodeGrant;
+    }
+
+    @Override
+    public boolean equals(final Object obj)
+    {
+        return obj instanceof ApplicationInfoImpl && this.id.equals(((ApplicationInfoImpl) obj).id);
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return this.description;
+    }
+
+    @Override
+    public String getIconId()
+    {
+        return this.iconId;
+    }
+
+    @Override
+    public String getIconUrl()
+    {
+        return this.iconId == null ? null
+                : "https://cdn.discordapp.com/app-icons/" + this.id + '/' + this.iconId + ".jpg";
```

Comment:
honestly idk, atm all 3 will return the image with a black background (even .png), not sure if that will change in the future